optimum/habana/transformers/trainer.py

```diff
@@ -142,6 +142,12 @@ def __init__(
             logger.info(f"No `GaudiTrainingArguments` passed, using `output_dir={output_dir}`.")
             args = GaudiTrainingArguments(output_dir=output_dir)
 
+        if args.half_precision_backend is not None and args.half_precision_backend == 'hpu_amp':
+            delattr(args, 'half_precision_backend')
+            self.use_hpu_amp = True
+        else:
+            self.use_hpu_amp = False
```

Comment:
```suggestion
        self.use_hpu_amp = False
        self.use_cpu_amp = False
        if args.bf16:
            if args.half_precision_backend == 'hpu_amp':
                self.use_hpu_amp = True
                delattr(args, 'half_precision_backend')
            else:
                self.use_cpu_amp = True
```
if we use `args.bf16`